src/index.ts

```diff
@@ -10,11 +10,11 @@ export { AST, RegExpParser, RegExpValidator }
  * @param options The options to parse.
  * @returns The AST of the regular expression.
  */
-export function parseRegExpLiteral(
-    source: string,
+export function parseRegExpLiteral<T>(
+    source: T extends RegExp ? RegExp : string,
     options?: RegExpParser.Options,
 ): AST.RegExpLiteral {
-    return new RegExpParser(options).parseLiteral(source)
+    return new RegExpParser(options).parseLiteral(source instanceof RegExp ? `/${source.source}/${source.flags}` : source)
```

Comment:
> `` `/${source.source}/${source.flags}` `` 

This is exact what `RegExp#toString()` does. I think you can use `String(source)` merely.